src/metabase/driver/mongo/query_processor.clj

```diff
@@ -390,14 +452,40 @@
        :collection source-table-name
        :mbql?      true})))
 
+;; omf: if value 'is' a ISODate, extract the date/time string and convert it to date object
+(defn- to-date-object [value]
+  (if (and (string? value) (clojure.string/starts-with? value "ISODate"))
+    (let [reg (re-seq #"ISODate\((.*?)\)" value)
```

Comment:
this regex doesn't look right, shouldn't it be something like `ISODate\((.+)\)` or even `ISODate\(([^)]+)\)`?